test/blogtoblog/BlogToBlogImport.spec.ts

```diff
@@ -24,19 +24,142 @@ const getImporter = (): BlogToBlogImporter => {
+            </div>
+          </div>
+        </div>
+      </main>`,
+      `<main>${div}${div}${div}
+        <div>
+          <table>
+            <tr><th>Another Block</th></tr>
+            <tr><td>cell11</td><td>cell12</td></tr>
+            <tr><td>cell21</td><td>cell22</td></tr>
+            <tr><td><img src="https://www.sample.com/image.jpeg"></td><td><a href="https://www.sample.com/">A link</a></td></tr>
+          </table>
+        </div>
+      </main>`);
+    test(
+      `<main>${div}${div}${div}
+        <div>
+        <div class="promotion">
+          <div>
+            <div><a href="https://blog.adobe.com/en/promotions/doc-cloud-education.html">https://blog.adobe.com/en/promotions/doc-cloud-education.html</a></div>
+          </div>
+        </div>
+      </main>`,
+      `<main>${div}${div}${div}
+        <div>
+          <table>
+            <tr><th>Promotion</th></tr>
+            <tr><td><a href="https://blog.adobe.com/en/promotions/doc-cloud-education.html">https://blog.adobe.com/en/promotions/doc-cloud-education.html</a></td></tr>
+          </table>
+        </div>
+      </main>`);
+  });
+});
+
+describe('BlogToBlogImporter#createTable tests', () => {
+  const test = (data: string[][], expected: string) => {
+    const { document } = (new JSDOM()).window;
+    const table = getImporter().createTable(data, document);
+    strictEqual(table.outerHTML, expected);
+  };
+
+  const div = '<div></div>'; // ignored div for the tests
+
+  it('convertBlocksToTables - basic tables', () => {
+    test(
+      [[]],
+      `<table><tr></tr></table>`);
+    test(
+      [['header']],
+      `<table><tr><th>header</th></tr></table>`);
+    test(
+      [['header'], ['cell']],
+      `<table><tr><th>header</th></tr><tr><td>cell</td></tr></table>`);
+    test(
+      [['header1', 'header2'], ['cell11', 'cell12'], ['cell21', 'cell22']],
+      `<table><tr><th>header1</th><th>header2</th></tr><tr><td>cell11</td><td>cell12</td></tr><tr><td>cell21</td><td>cell22</td></tr></table>`);
+    // TODO deal with colspan ?
+    test(
+      [['header1'], ['cell11', 'cell12'], ['cell21', 'cell22']],
+      `<table><tr><th>header1</th></tr><tr><td>cell11</td><td>cell12</td></tr><tr><td>cell21</td><td>cell22</td></tr></table>`);
+  });
+
+  it('convertBlocksToTables - deals with Elements', () => {
+    const { document } = (new JSDOM()).window;
+
+    const img = document.createElement('img');
+    img.src = 'https://www.sample.com/image.jpeg';
+
+    const a = document.createElement('a');
+    a.href = 'https://www.sample.com/';
+    a.innerHTML = 'A link';
+
+    test(
+      [['header'], [ img ]],
+      `<table><tr><th>header</th></tr><tr><td><img src="https://www.sample.com/image.jpeg"></td></tr></table>`);
+    test(
+      [['header'], [ img, a, 'some text' ]],
+      `<table><tr><th>header</th></tr><tr><td><img src="https://www.sample.com/image.jpeg"></td><td><a href="https://www.sample.com/">A link</a></td><td>some text</td></tr></table>`);
```

Comment:
I'm fairly certain the business website is setup to read image tables as a two-row table:
```
<table>
  <tr>
    <th>block name</th>
  </tr>
  <tr>
    <td>
      <img />
      <a>link</a>
      <p><em>caption</em></p>
    </td>
  </tr>
</table>```

Reply:
Maybe this is just part of the bigger conversation. Where do we make this structure decision -- here or the blog?

Reply:
These are pure tests, not necessarily related to "real" structure. But were you are right is that your structure must be achievable. I'll add a test for it.